stdlib/char.mli

```diff
@@ -53,6 +57,134 @@ val equal: t -> t -> bool
+
+  val is_alphanum : char -> bool
+  (** [is_alphanum c] is {!is_letter}[ c || ]{!is_digit}[ c]. *)
+
+  val is_white : char -> bool
+  (** [is_white c] is [true] if and only if [c] is an ASCII white space
+      character, that is one of
+      tab ['\t'] ([0x09]), newline ['\n'] ([0x0A]),
+      vertical tab ([0x0B]), form feed ([0x0C]),
+      carriage return ['\r'] ([0x0D]) or space [' '] ([0x20]),  *)
+
+  val is_blank : char -> bool
+  (** [is_blank c] is [true] iff [c] is an ASCII blank character,
+      that is either space [' '] ([0x20]) or tab ['\t'] ([0x09]). *)
+
+  val is_graphic : char -> bool
+  (** [is_graphic c] is [true] iff [c] is an ASCII graphic
+      character that is a byte in the range \[[0x21];[0x7E]\]. *)
+
+  val is_print : char -> bool
+  (** [is_print c] is {!is_graphic}[ c || c = ' ']. *)
+
+  val is_control : char -> bool
+  (** [is_control c] is [true] iff [c] is an ASCII control character,
+      that is a byte in the range \[[0x00];[0x1F]\] or [0x7F]. *)
+
+  (** {1:decimal_digits Decimal digits} *)
+
+  val is_digit : char -> bool
+  (** [is_digit c] is [true] iff [c] is an ASCII digit
+      ['0'] to ['9'], that is a byte in the range \[[0x30];[0x39]\]. *)
+
+  val digit_to_int : char -> int
+  (** [digit_to_int c] is the numerical value of a digit
+      that satisfies {!is_digit}. Raises [Invalid_argument] if
+      {!is_digit}[ c] is [false]. *)
+
+  val digit_of_int : int -> char
+  (** [digit_of_int n] is an ASCII digit for the decimal
+      digit [n mod 10]. *)
+
+  (** {1:hex_digits Hexadecimal digits} *)
+
+  val is_hex_digit : char -> bool
+  (** [is_hex_digit c] is [true] iff [c] is an ASCII hexadecimal
+      digit ['0'] to ['9'], ['a'] to ['f'] or ['A'] to ['F'],
+      that is a byte in one of the ranges \[[0x30];[0x39]\],
+      \[[0x41];[0x46]\], \[[0x61];[0x66]\]. *)
+
+  val hex_digit_to_int : char -> int
+  (** [hex_digit_to_int c] is the numerical value of a digit that
+      satisfies {!is_hex_digit}. Raises [Invalid_argument] if
+      {!is_hex_digit}[ c] is [false]. *)
+
+  val lower_hex_digit_of_int : int -> char
+  (** [lower_hex_digit n] is an lowercase ASCII hexadecimal digit for
+      the integer [n] truncated to its lowest 4 bits. *)
+
+  val upper_hex_digit_of_int : int -> char
+  (** [upper_hex_digit_of_int n] is an uppercase ASCII hexadecimal
+      digit for the integer [n] truncated to its lowest 4 bits. *)
+
+  (** {1:casing Casing transforms} *)
+
+  val uppercase : char -> char
+  (** [uppercase c] is [c] with ASCII characters ['a'] to ['z'] respectively
+      mapped to uppercase characters ['A'] to ['Z']. Other characters are left
+      untouched. *)
+
+  val lowercase : char -> char
+  (** [lowercase c] is [c] with ASCII characters ['A'] to ['Z'] respectively
+      mapped to lowercase characters ['a'] to ['z']. Other characters are
+      left untouched. *)
+end
+
+val lowercase_ascii : char -> char
+(** Use the equivalent {!Ascii.lowercase} instead.
```

Comment:
Usually we only add deprecation warnings after a few release. Otherwise people who need to support older OCaml versions get the warning without being able to act on it except by silencing it which defeats the purpose. 

At some point I suggested to have a table of warning introduction which the release manager would have to consult. Now that I think of it, in fact we simply need a smarter `@@ocaml.deprecated` that starts warning at a given version (and a policy on after how many number version numbers after the workaround was introduced, the warning becomes active).